Sources/Network/NetworkUpload.swift

```diff
@@ -63,7 +72,31 @@ open class NetworkUploadProcedure<Session: URLSessionTaskFactory>: Procedure, Re
             strongSelf.finish()
         }
 
+        task?.addObserver(self, forKeyPath: #keyPath(URLSessionTaskProgressProtocol.countOfBytesExpectedToSend), options: [.new, .old], context: &observerContext)
+        task?.addObserver(self, forKeyPath: #keyPath(URLSessionTaskProgressProtocol.countOfBytesSent), options: [.new, .old], context: &observerContext)
+
         task?.resume()
     }
 
+    override open func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        guard context == &observerContext else {
+            super.observeValue(forKeyPath: keyPath, of: object, change: change, context: context)
+            return
+        }
+
+        if let keyPath = keyPath, let task = object as? Session.UploadTask {
+            switch keyPath {
+            case #keyPath(URLSessionTaskProgressProtocol.countOfBytesExpectedToSend):
+                progress.totalUnitCount = task.countOfBytesExpectedToSend
+            case #keyPath(URLSessionTaskProgressProtocol.countOfBytesSent):
+                progress.completedUnitCount = task.countOfBytesSent
+            case #keyPath(URLSessionTaskProgressProtocol.countOfBytesExpectedToReceive):
+                progress.totalUnitCount = task.countOfBytesExpectedToReceive
+            case #keyPath(URLSessionTaskProgressProtocol.countOfBytesReceived):
```

Comment:
It seems like this case is unnecessary for `NetworkUploadProcedure`? (It's not being registered for KVO.)